File: 10/code-shapes-use.java
/* #Полиморфизм подтипов: Circle можно рассматривать как Shape.# */
Shape s = new Circle(new Point(1, 1), 5);

/* #Динамическая диспетчеризация: вызывается Circle.draw(), а не Shape.draw().# */
s.draw();

public abstract class Shape {
	/*...*/

	public void drawAndFill() {
		/* #Открытая рекурсия: при вызове s.drawAndFill() будут вызваны# */
		/* #методы Circle.draw() и Circle.fill().# */
		this.draw();
		this.fill();
	}
}

File: 10/code-shapes-use.js
var shape = new Circle({'x': 1, 'y': 1}, 5);

/* #Работает, несмотря на отсутствие методов .draw() и .fill() в Shape.# */
shape.drawAndFill();
/* #Заменяет метод для конкретного объекта.# */
shape.draw = function() { /*...*/ };
/* #Заменяет метод для всех объектов, создаваемых с помощью new Circle(...).# */
Circle.prototype.draw = function() { /*...*/ };

var otherShape = {
	'draw': function() { alert('Drawn!'); }
	'fill': function() { alert('Filled!'); }
	'drawAndFill': Shape.prototype.drawAndFill
};
/* #Duck typing: работает, несмотря на отсутствие явного наследования.# */
otherShape.drawAndFill();
